packages/flutter/lib/src/gestures/mouse_cursor.dart

```diff
@@ -0,0 +1,332 @@
+/// cursors on platforms. To assign widgets with system cursors, use the objects
+/// defined in [SystemMouseCursors] instead.
+///
+/// See also:
+///
+///  * [SystemMouseCursors], which contains usable [MouseCursor] objects that
+///    correspond to values of this type.
+///  * [MouseCursorPlatformDelegate], which uses this type to define how
+///    system cursors are implemented on platforms.
+enum MouseCursorSystemShape {
+  /// The shape that corresponds to [SystemCursors.none].
+  none,
+
+  /// The shape that corresponds to [SystemCursors.basic].
+  ///
+  /// This shape must be implemented by all platforms.
+  basic,
+
+  /// The shape that corresponds to [SystemCursors.click].
+  click,
+
+  /// The shape that corresponds to [SystemCursors.text].
+  text,
+
+  /// The shape that corresponds to [SystemCursors.forbidden].
+  forbidden,
+
+  /// The shape that corresponds to [SystemCursors.grab].
+  grab,
+
+  /// The shape that corresponds to [SystemCursors.grabbing].
+  grabbing,
+}
+
+/// Details for [MouseCursorPlatformDelegate.activateSystemCursor], such as the
+/// target device and the system cursor shape.
+@immutable
+class MouseCursorPlatformActivateSystemCursorDetails {
+  /// Create details for a [MouseCursorPlatformDelegate.activateSystemCursor]
+  /// call.
+  ///
+  /// All parameters must not be null.
+  const MouseCursorPlatformActivateSystemCursorDetails({
+    @required this.device,
+    @required this.systemShape,
+  }) : assert(device != null), assert(systemShape != null);
+
+  /// The pointer device that should change cursor.
+  final int device;
+
+  /// The kind of system cursor that should change to.
+  final MouseCursorSystemShape systemShape;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != MouseCursorPlatformActivateSystemCursorDetails)
+      return false;
+    final MouseCursorPlatformActivateSystemCursorDetails typed = other;
+    return typed.device == device && typed.systemShape == systemShape;
+  }
+
+  @override
+  int get hashCode => hashValues(device, systemShape);
+
+  @override
+  String toString() {
+    return '$runtimeType(device: $device, systemShape: $systemShape)';
+  }
+}
+
+/// An interface for the operations that a [MouseCursor] can use to control the
+/// platform.
+///
+/// This interface is implemented by each platform that supports mouse cursor.
+///
+/// See also:
+///
+///  * [MouseCursor], whose subclasses and methods use this class to perform
+///    operations.
+///  * [MouseCursorManager], which takes this class as a parameter.
+abstract class MouseCursorPlatformDelegate {
+  /// Create a [MouseCursorPlatformDelegate].
+  const MouseCursorPlatformDelegate();
+
+  /// Asks the platform to change the cursor of `device` to the system cursor
+  /// specified by `systemShape`.
+  ///
+  /// It resolves to `true` if the operation is successful, `false` if the
+  /// operation is unsupported by the platform, or rejects to error if the
```

Comment:
From reading the code elsewhere: "unsupported" may mean that this type of cursor is not available in the platform? Maybe document that?